operator/v1/types_ingress.go

```diff
@@ -792,6 +806,7 @@ const (
 // IngressController, and represents the type and any additional configuration
 // for a specific type.
 // +union
+// +kubebuilder:validation:XValidation:rule="!has(self.loadBalancer) || !has(self.loadBalancer.scope) || self.loadBalancer.scope != 'Internal' || !has(self.loadBalancer.providerParameters) || !has(self.loadBalancer.providerParameters.aws) || !has(self.loadBalancer.providerParameters.aws.networkLoadBalancer) || !has(self.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations)",message="eipAllocations can't be provided when the scope is Internal."
```

Comment:
Any reason you didn't put this on `LoadBalancerStrategy` to simply the CEL? Everything starts with `self.loadBalancer`.

Reply:
sure made the changes